adam/curriculum_to_html.py

```diff
@@ -398,9 +398,35 @@ def dfs_walk(node):
                     (relation_prefix, relation_suffix) = compute_arrow(
                         relation, static_relations, first_frame_relations
                     )
-                    output_text.append(
-                        f"\t\t\t\t\t\t<li>{relation_prefix}{relation}{relation_suffix}</li>"
-                    )
+                    # if matching smallerThan/biggerThan relations exist, give as single relation
+                    opposite_relations = {
+                        SMALLER_THAN: BIGGER_THAN,
+                        BIGGER_THAN: SMALLER_THAN,
+                    }
+                    single_size_relation = None
+                    if relation.relation_type in opposite_relations:
+                        if any(
```

Comment:
Note this will be quadratic time in the total number of relations, which can sometimes be large.  You could index the relations by tuples of their slot pairs to provide O(1) lookup